src/connectors/range/__tests__/connectRange-test.ts

```diff
@@ -0,0 +1,2151 @@
+        attribute,
+      });
+
+      const { refine } = widget.getWidgetRenderState(
+        createInitOptions({ helper })
+      );
+
+      refine([10, 490]);
+
+      expect(helper.getNumericRefinement(attribute, '>=')).toEqual([10]);
+      expect(helper.getNumericRefinement(attribute, '<=')).toEqual([490]);
+      expect(helper.state.removeNumericRefinement).toHaveBeenCalledWith(
+        attribute
+      );
+      expect(helper.search).toHaveBeenCalled();
+    });
+
+    it('expect to refine when values are in range', () => {
+      const helper = createHelper();
+      const widget = connectRange(rendering)({ attribute });
+
+      helper.setState(
+        widget.getWidgetSearchParameters!(helper.state, { uiState: {} })
+      );
+
+      const { refine } = widget.getWidgetRenderState(
+        createRenderOptions({
+          helper,
+          results: createFacetStatsResults({
+            min: 0,
+            max: 500,
+            helper,
+            attribute,
+          }),
+        })
+      );
+
+      refine([10, 490]);
+
+      expect(helper.getNumericRefinement(attribute, '>=')).toEqual([10]);
+      expect(helper.getNumericRefinement(attribute, '<=')).toEqual([490]);
+      expect(helper.state.removeNumericRefinement).toHaveBeenCalledWith(
+        attribute
+      );
+      expect(helper.search).toHaveBeenCalled();
+    });
+
+    it('expect to refine when values are parsable integer', () => {
+      const helper = createHelper();
+      const widget = connectRange(rendering)({ attribute });
+
+      helper.setState(
+        widget.getWidgetSearchParameters!(helper.state, { uiState: {} })
+      );
+
+      const { refine } = widget.getWidgetRenderState(
+        createRenderOptions({
+          helper,
+          results: createFacetStatsResults({
+            min: 0,
+            max: 500,
+            helper,
+            attribute: 'price',
+          }),
+        })
+      );
+
+      // @ts-ignore
+      refine(['10', '490']);
+
+      expect(helper.getNumericRefinement(attribute, '>=')).toEqual([10]);
+      expect(helper.getNumericRefinement(attribute, '<=')).toEqual([490]);
+      expect(helper.state.removeNumericRefinement).toHaveBeenCalled();
+      expect(helper.search).toHaveBeenCalled();
+    });
+
+    it('expect to refine when values are parsable float', () => {
+      const helper = createHelper();
+      const widget = connectRange(rendering)({ attribute });
+
+      helper.setState(
+        widget.getWidgetSearchParameters!(helper.state, { uiState: {} })
+      );
+
+      const { refine } = widget.getWidgetRenderState(
+        createRenderOptions({
+          helper,
+          results: createFacetStatsResults({
+            min: 0,
+            max: 500,
+            helper,
+            attribute: 'price',
+          }),
+        })
+      );
+
+      // @ts-ignore
+      refine(['10.50', '490.50']);
+
+      expect(helper.getNumericRefinement(attribute, '>=')).toEqual([10]);
```

Comment:
it'd be nice to have a little comment about floor for min and ceil for max.

Reply:
36e87c07